lib/common/socket.c

```diff
@@ -682,6 +682,27 @@ int h2o_socket_get_ssl_cipher_bits(h2o_socket_t *sock)
     return sock->ssl != NULL ? SSL_get_cipher_bits(sock->ssl->ssl, NULL) : 0;
 }
 
+
+h2o_iovec_t h2o_socket_log_ssl_session_id(h2o_socket_t *sock, h2o_mem_pool_t *pool)
+{
+    h2o_iovec_t key;
+    unsigned id_len;
+    const unsigned char *id;
+
+    switch (sock->ssl->handshake.server.async_resumption.state) {
+        case ASYNC_RESUMPTION_STATE_COMPLETE:
+            id = SSL_SESSION_get_id(sock->ssl->ssl->session, &id_len);
+            #define BASE64_LENGTH(len) (((len) + 2) / 3 * 4 + 1)
```

Comment:
How about promoting this macro into a inline function defined in string_.h?

Regarding the name of the inline function, I think it should be named something like `h2o_base64_encode_capacity` or something that reflect the fact that the returned value is not the exact number of bytes required to store the output. The returned value always takes trailing `=` and NUL character into account.